runtime-modules/proposals/engine/src/lib.rs

```diff
@@ -589,13 +590,35 @@ impl<T: Trait> Module<T> {
 
     /// Resets voting results for active proposals.
     /// Possible application includes new council elections.
-    pub fn reset_active_proposals() {
+    pub fn reset_votes_for_active_proposals() {
         <ActiveProposalIds<T>>::iter().for_each(|(proposal_id, _)| {
+            Self::reset_proposal_votes(proposal_id);
+        });
+    }
+
+    // Resets votes for a single proposal.
+    fn reset_proposal_votes(proposal_id: T::ProposalId) {
+        <Proposals<T>>::mutate(proposal_id, |proposal| {
+            proposal.reset_proposal_votes();
+            <VoteExistsByProposalByVoter<T>>::remove_prefix(&proposal_id);
+        });
+    }
+
+    /// Reactivate proposals with pending constitutionality.
+    /// Possible application includes new council elections.
+    pub fn reactivate_pending_constitutionality_proposals() {
+        <PendingConstitutionalityProposalIds<T>>::iter().for_each(|(proposal_id, _)| {
```

Comment:
Can this be done more functionally, just computing new proposals and reinsert them into `Proposals`.

Reply:
It does several activities like increasing the proposal counter. I'm not sure I understand your note, we should probably discuss it.

Reply:
I refactored it after our discussion.